tests/Unit/Domain/FunctionsOfTime/Test_QuaternionFunctionOfTime.cpp

```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cmath>
+
+#include "DataStructures/DataVector.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/FunctionsOfTime/QuaternionFunctionOfTime.hpp"
+
+SPECTRE_TEST_CASE("Unit.Domain.FunctionsOfTime.QuaternionFuntionOfTime",
+                  "[Unit][Domain]") {
+  INFO("Test quaternion function of time") {
+    double t = 0.0;
+    double expir_time = 0.5;
+    const double omega_z = 1.3333;
+    DataVector init_omega{0.0, 0.0, omega_z};
+    // Construct PiecewisePolynomial
+    domain::FunctionsOfTime::PiecewisePolynomial<2> pp{
+        t,
+        std::array<DataVector, 3>{
+            {init_omega, DataVector{3, 0.0}, DataVector{3, 0.0}}},
+        expir_time};
+
+    // Update PiecewisePolynomial with 0 2nd derivative so it's constant
+    const double increment = 0.5;
+    for (int i = 0; i < 15; i++) {
+      t += increment;
+      expir_time += increment;
+      pp.update(t, DataVector{3, 0.0}, expir_time);
+    }
+    t = 0.0;
+    expir_time = 0.5;
+    DataVector init_quat{1.0, 0.0, 0.0, 0.0};
+    // Construct QuaternionFunctionOfTime
+    domain::FunctionsOfTime::QuaternionFunctionOfTime<2> qfot{
+        t, std::array<DataVector, 1>{init_quat}, make_not_null(&pp),
+        expir_time};
+
+    // Get the quaternion and 2 derivatives at a certain time. This call will
+    // automatically update the stored quaternions to match the stored omegas
+    double check_time = 5.398;
+    const std::array<DataVector, 3> quat_func_and_2_derivs =
+        qfot.func_and_2_derivs(check_time);
+
+    // Analytic solution for constant omega
```

Comment:
Please add a test for non-constant omega as well as this test for constant omega. Perhaps omega linear or quadratic in time. [recall that an early version of the SpEC QuatFvt was correct for constant omega but wrong for non-constant omega].